### ISSUE TITLE
`NoSuchElementException` when `Ctrl + hover` over a type

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#29440]
Consider the single file below:
```ballerina
import  ballerina/http;

listener  http:Listener  ep0  = new (9090, config  = {host: localhost});

 service  /v1  on  ep0  {
        resource  function  get  pets(int?  'limit)  returns  Pets|Error {
    }
        resource  function  post  pets()  returns  http:Created|Error {
    }
        resource  function  get  pets/[string  petId]()  returns  Pet|Error {
    }
}
```
When I press `Ctrl` and hover over `http:Created`, getting the following exception:
```
[Error - 2:26:36 PM] Operation 'text/definition' failed! {uri: '/c%3A/Users/Pubudu/Projects/openapi/foo-service.bal', [8:62], error: 'No value present'}
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.orElseThrow(Optional.java:382)
	at org.ballerinalang.langserver.util.definition.DefinitionUtil.getLocation(DefinitionUtil.java:82)
	at org.ballerinalang.langserver.util.definition.DefinitionUtil.getDefinition(DefinitionUtil.java:61)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$definition$3(BallerinaTextDocumentService.java:257)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by nadeeshaan** [comment]
Goto definition is not supported yet for the modules instead the ones are within a given package.

**Comment by nadeeshaan** [comment]
Related Issue #29558

**Comment by nadeeshaan** [comment]
Fixed with #29744. Hence closing the issue
